src/DotNet.Testcontainers.Tests/Unit/Containers/Unix/Database/CouchbaseTestcontainerTest.cs

```diff
@@ -1,106 +1,118 @@
-      var updatedCustomer = (await collection.GetAsync("customer-id-1")).ContentAs<Customer>();
-      Assert.True(updatedCustomer.Name == "Onur");
-      Assert.True(updatedCustomer.Age == 30);
-
-      //Delete
-      await collection.RemoveAsync("customer-id-1");
-      var existsResult = await collection.ExistsAsync("customer-id-1");
-      Assert.False(existsResult.Exists);
+      // Given
+      const string bucketName = nameof(this.ShouldContainCreatedBucket);
+
+      await this.couchbaseFixture.Container.CreateNewBucket(bucketName);
+
+      await using (var cluster = await Cluster.ConnectAsync(
+        this.couchbaseFixture.Container.ConnectionString,
+        this.couchbaseFixture.Container.Username,
+        this.couchbaseFixture.Container.Password))
+      {
+        // When
+        var buckets = await cluster.Buckets.GetAllBucketsAsync();
+
+        // Then
+        Assert.True(buckets.ContainsKey(bucketName));
+      }
     }
 
     [Fact]
-    public async Task CreateNewBucketTest()
+    public async Task ShouldNotContainFlushedBucket()
     {
-      //Given
-      await this.couchbase.Container.CreateNewBucket("dotnet-testcontainers",128);
+      // Given
+      const string bucketName = "Example";
 
-      //When
-      var cluster = await Cluster.ConnectAsync(this.couchbase.Container.ConnectionString, "Administrator", "password");
-      var bucket = await cluster.Buckets.GetAllBucketsAsync();
+      await this.couchbaseFixture.Container.CreateNewBucket(bucketName);
 
-      //Then
-      Assert.True(bucket.ContainsKey("dotnet-testcontainers"));
-    }
+      await using (var cluster = await Cluster.ConnectAsync(
+        this.couchbaseFixture.Container.ConnectionString,
+        this.couchbaseFixture.Container.Username,
+        this.couchbaseFixture.Container.Password))
+      {
+        await cluster.QueryAsync<long>($"CREATE PRIMARY INDEX `#primary` ON `{bucketName}`");
 
-    [Fact]
-    public async Task FlushBucketTest()
-    {
-      //Given
-      await this.couchbase.Container.CreateNewBucket("Example",128);
+        await using (var bucket = await cluster.BucketAsync(bucketName))
+        {
+          var collection = bucket.DefaultCollection();
 
-      var cluster = await Cluster.ConnectAsync(this.couchbase.Container.ConnectionString, "Administrator", "password");
-      await cluster.QueryAsync<long>("CREATE PRIMARY INDEX `#primary` ON `Example`");
-      var bucket = await cluster.BucketAsync("Example");
-      var collection = bucket.DefaultCollection();
+          await collection.InsertAsync("1", new { Phone = "123456" });
 
-      await collection.InsertAsync(
-        "1",
-        new { Phone = "123456" }
-      );
+          // When
+          await this.couchbaseFixture.Container.FlushBucket(bucketName);
 
-      //When
-      await this.couchbase.Container.FlushBucket("Example");
+          // Then
+          var result = await cluster.QueryAsync<long>($"SELECT * FROM {bucketName}"); // Don't do this.
```

Comment:
It's possible to execute count query. Or maybe better to fetch only document which has id "1". Do you have any suggestion?

Reply:
I don't know a lot about couchbase. But it looks like a regular SQL query. I'm pretty sure this will work. Should be some improvement for the next issue / pr.